rest/rest-smartproxy/src/main/java/org/ow2/proactive_grid_cloud_portal/smartproxy/RestSmartProxyImpl.java

```diff
@@ -312,44 +303,66 @@ public void addEventListener(SchedulerEventListenerExtended listener, boolean my
-                    addfileSelection(is.getInputFiles(), includes, excludes);
-                }
+            if (remotePath == null) {
+                // NOTE: only works for USERSPACE urls
+                throw new IllegalArgumentException("Could not extract remote path using inputSpace=" + inputSpaces +
+                                                   " and userSpace=" + userSpaces);
             }
+            String jname = job.getName();
+            logger.debug("Pushing files for job " + jname + " from " + localInputFolderPath + " to " + remotePath);
+            TaskFlowJob tfj = job;
+            for (Task t : tfj.getTasks()) {
+                logger.debug("Pushing files for task " + t.getName());
+                List<String> includes = Lists.newArrayList();
+                List<String> excludes = Lists.newArrayList();
+
+                List<InputSelector> inputFilesList = t.getInputFilesList();
+
+                if (inputFilesList != null) {
+                    for (InputSelector is : inputFilesList) {
+                        addfileSelection(is.getInputFiles(), includes, excludes);
+                    }
+                }
 
-            LocalDirSource source = new LocalDirSource(localInputFolderPath);
-            source.setIncludes(includes);
-            source.setExcludes(excludes);
+                LocalDirSource source = new LocalDirSource(localInputFolderPath);
+                source.setIncludes(includes);
+                source.setExcludes(excludes);
 
-            RemoteDestination dest = new RemoteDestination(USER, remotePath);
-            restDataSpaceClient.upload(source, dest);
+                RemoteDestination dest = new RemoteDestination(USER, remotePath);
+                restDataSpaceClient.upload(source, dest);
+            }
+            logger.debug("Finished push operation from " + localInputFolderPath + " to " + remotePath);
+
+        } catch (Throwable error) {
+            throw Throwables.propagate(error);
         }
-        logger.debug("Finished push operation from " + localInputFolderPath + " to " + remotePath);
 
         return true;
     }
 
+    private String extractRemotePath(List<String> userSpaces, List<String> inputSpaces) {
+        String remotePath = null;
+        for (int i = 0; i < inputSpaces.size(); i++) {
+            if (i >= userSpaces.size()) {
+                break;
+            }
+            String inputSpace = inputSpaces.get(i);
+            String userSpace = userSpaces.get(i);
+
+            if (inputSpace.startsWith(userSpace)) {
+                remotePath = inputSpace.substring(userSpace.length() + (userSpace.endsWith("/") ? 0 : 1));
+                break;
+            }
+        }
+        return remotePath;
+    }
+
     @Override
     protected void downloadTaskOutputFiles(AwaitedJob awaitedjob, String jobId, String taskName, String localFolder)
             throws NotConnectedException, PermissionException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)